prism_compile.c

```diff
@@ -3714,7 +3717,50 @@ pm_compile_call(rb_iseq_t *iseq, const pm_call_node_t *call_node, LINK_ANCHOR *c
         PUSH_INSN(ret, location, splatkw);
     }
 
-    PUSH_SEND_R(ret, location, method_id, INT2FIX(orig_argc), block_iseq, INT2FIX(flags), kw_arg);
+    LABEL *not_basic_new = NEW_LABEL(location.line);
+    LABEL *not_basic_new_finish = NEW_LABEL(location.line);
+
+    bool inline_new = ISEQ_COMPILE_DATA(iseq)->option->specialized_instruction &&
+        method_id == rb_intern("new") &&
+        call_node->block == NULL &&
+        !(flags & VM_CALL_ARGS_BLOCKARG);
```

Comment:
I think we don't need this line because in Prism full blocks and block arguments are both attached at `call_node->block`, but in parse.y the full blocks surround the calls and the regular ones are passed as args_add_block.